ThunkLibs/Generator/analysis.cpp

```diff
@@ -184,9 +221,14 @@ void AnalysisAction::ParseInterface(clang::ASTContext& context) {
             type = type->getLocallyUnqualifiedSingleStepDesugaredType();
 
             if (type->isFunctionPointerType() || type->isFunctionType()) {
-                funcptr_types.insert(type.getTypePtr());
+                funcptr_types["TODO_FUNC_NAME_FOR_ANNOTATED_TYPES_" + type.getAsString()] = std::pair { type.getTypePtr(), no_param_annotations };
```

Comment:
Revisit this. Adding a fixed prefix hasn't caused any problems so far, but to be on the safe side it may indeed be needed to add the full function name.

Reply:
Actually this is safe, since the type name for a function pointer type will never conflict with a function name.